lib/matplotlib/backends/backend_pgf.py

```diff
@@ -23,37 +23,52 @@
     _create_pdf_info_dict, _datetime_to_pdf)
 from matplotlib.path import Path
 from matplotlib.figure import Figure
+from matplotlib.font_manager import FontProperties
 from matplotlib._pylab_helpers import Gcf
 
 _log = logging.getLogger(__name__)
 
 
+_DOCUMENTCLASS = r"\documentclass{article}"
+
+
 # Note: When formatting floating point values, it is important to use the
 # %f/{:f} format rather than %s/{} to avoid triggering scientific notation,
 # which is not recognized by TeX.
 
 def _get_preamble():
     """Prepare a LaTeX preamble based on the rcParams configuration."""
+    font_size_pt = FontProperties(
+        size=mpl.rcParams["font.size"]
+    ).get_size_in_points()
     return "\n".join([
         # Remove Matplotlib's custom command \mathdefault.  (Not using
         # \mathnormal instead since this looks odd with Computer Modern.)
         r"\def\mathdefault#1{#1}",
         # Use displaystyle for all math.
         r"\everymath=\expandafter{\the\everymath\displaystyle}",
+        # Set up font sizes to match font.size setting.
+        r"\IfFileExists{scrextend.sty}{",
```

Comment:
Please document that (IIUC?) the effecf of scrextend (koma-script) here is that all fontsize selectors (HUGE, Large, large, etc.) will be correctly sized as well.

Reply:
Yes, when passing the package option `fontsize=`, that’s the effect of `scrextend`. Is a comment enough or do you mean user-facing documentation?

Reply:
A comment is enough.

Reply:
Done!

Reply:
Is there a reason why this `scrextend` code is not guarded by the `pgf.rcfonts` flag too?  I think the `scrextend` change is the only thing that I had some opposition to because, unless I'm mistaken, I think it can break consistency.  If it's only used when `pgf.rcfonts=False`, then I think it would at least have an easy workaround.

Reply:
It seems more consistent to me to always set the document font size to a well-defined value, rather than having this depend on `pgf.rcfonts`. What do you mean by “break consistency”? There shouldn’t be any difference in plot outputs, since font sizes are always set locally as well.

Reply:
I provided an example here due to space limitations: https://github.com/matplotlib/matplotlib/pull/26893#issuecomment-2094003950

Please let me know if I made a mistake.  It's very likely that I am just misunderstanding something and this is a non-issue.